cmd/zt_sync_cmd_test.go

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2017 Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	chk "gopkg.in/check.v1"
+)
+
+//Test dfs endpoints are cooked to blob endpoints
+func (s *cmdIntegrationSuite) TestSyncS2SWithDFS(c *chk.C) {
+	src := "https://myaccount1.dfs.core.windows.net/container1/"
+	dst := "https://myaccount2.dfs.core.windows.net/container2/"
+
+	raw := getDefaultSyncRawInput(src, dst)
+
+	cooked, err := raw.cook()
+
+	c.Assert(err, chk.IsNil)
+	c.Assert(cooked.fromTo, chk.Equals, common.EFromTo.BlobBlob())
```

Comment:
Our testing should really be more in depth than this for something like this (even though such operations by all means should work as expected). You should set up a job like this in the declarative test suite, just on the off-chance something is not happy.

Reply:
That would mean we're not confident in other code, testing or both. Unless there is a reason other than to be on safer side, I'd be wary against adding test to a suite which is already overloaded.

Reply:
+1 on adding a test to the new E2E framework (can be done after merging PR), the new framework is super fast and would give more confidence on this scenario.

Reply:
Agreed. Let's do an AllS2S() or AllPairs() test for DFS endpoint for more confidence in the scenario.

Reply:
E2E tests require implementation of resource manages for Gen2. That would be a significant feature in test-infra, and I dont want to include it here. The one in this file will give us similar confidence, we still execute whole copy/sync. Resource managers for Gen2 will be taken care of.